velero-plugins/imagestream/backup.go

```diff
@@ -34,6 +34,84 @@ func (p *BackupPlugin) Execute(item runtime.Unstructured, backup *v1.Backup) (ru
 
 	if backup.Labels[common.MigrationApplicationLabelKey] != common.MigrationApplicationLabelValue{
 		p.Log.Info("[is-backup] skipping imagestream plugin since backup is not part of CAM")
+		im := imagev1API.ImageStream{}
+		itemMarshal, _ := json.Marshal(item)
+		json.Unmarshal(itemMarshal, &im)
+		annotations := im.Annotations
+		if annotations == nil {
+			annotations = make(map[string]string)
+		}
+		internalRegistry := annotations[common.BackupRegistryHostname]
+		backupRegistry, err := getRoute(backup.Namespace, backup.Spec.StorageLocation, common.ConfigMap)
```

Comment:
This is the unique bits for non CAM case.

`internalRegistry` is populated the same way for CAM vs non-CAM. We want to have a condiitonal at the top:
```
if CAM {
  backupRegistry := <get coords from annotation>
} else {
  backupRegistry := getRoute()
}

Reply:
Then reuse copyImage code bits for CAM and non-CAM envs to not duplicate code.

Reply:
Actually, it might be better to use the annotation for the backupRegistry for both, and in the common plugin, include this:
```
if !CAM {
  // Set annotation to the result of the getRoute call
}
```
This makes the plugin execute calsl easier to read and isolates special handling of the registry location to one place in the code (in case it's needed in more than one location in the plugins).

Reply:
+1 I think that's a great idea